my_game.py

```diff
@@ -352,11 +371,192 @@ def on_mouse_press(self, x: int, y: int, button: int, modifiers: int):
+            text="Change Thrust Key: " + SettingsView.id_to_key[CONFIG["PLAYER_THRUST_KEY"]], width=300)
+        self.v_box.add(self.change_player_thrust_key_button.with_space_around(bottom=20))
+
+        self.change_player_fire_key_button = arcade.gui.UIFlatButton(
+            text="Change Fire Key: " + SettingsView.id_to_key[CONFIG["PLAYER_FIRE_KEY"]], width=300)
+        self.v_box.add(self.change_player_fire_key_button.with_space_around(bottom=20))
+
+        self.change_player_turn_right_key_button = arcade.gui.UIFlatButton(
+            text="Change Turn Right Key: " + SettingsView.id_to_key[CONFIG["PLAYER_TURN_RIGHT_KEY"]], width=300)
+        self.v_box.add(self.change_player_turn_right_key_button.with_space_around(bottom=20))
+
+        self.change_player_turn_left_key_button = arcade.gui.UIFlatButton(
+            text="Change Turn Left Key: " + SettingsView.id_to_key[CONFIG["PLAYER_TURN_LEFT_KEY"]], width=300)
+        self.v_box.add(self.change_player_turn_left_key_button.with_space_around(bottom=20))
+
+        self.v_box.add(self.settings_guide_1.with_space_around(bottom=20))
+
+        # Assign click functions to buttons
+        self.reset_settings_button.on_click = self.on_click_reset
+        
+        self.change_player_thrust_key_button.on_click = self.on_click_change_player_thrust_key
+        
+        self.change_player_fire_key_button.on_click = self.on_click_change_player_fire_key
+        
+        self.change_player_turn_right_key_button.on_click = self.on_click_change_turn_right_key
+        
+        self.change_player_turn_left_key_button.on_click = self.on_click_change_turn_left_key
+        
+        # Background Color
+        arcade.set_background_color(SCREEN_COLOR)
+        
+        # Add layout to UI Manager
+        self.manager.add(
+            arcade.gui.UIAnchorWidget(
+                anchor_x="center_x",
+                anchor_y="center_y",
+                child=self.v_box)
+        )
+
+    def on_click_reset(self, event):
+        # Delete edited config file
+        try:
+            os.remove("my_game_edit.toml")
+        except FileNotFoundError:
+            pass
+
+        # Reset CONFIG dict
+        with open('my_game.toml', 'rb') as fp:
+            CONFIG_reset = tomli.load(fp)
+        for k in CONFIG_reset:
+                CONFIG[k] = CONFIG_reset[k]
+        self.changed_settings = {}
+
+        self.change_player_thrust_key_button.text = "Change Thrust Key: " + SettingsView.id_to_key[CONFIG["PLAYER_THRUST_KEY"]]
+        self.change_player_fire_key_button.text = "Change Fire Key: " + SettingsView.id_to_key[CONFIG["PLAYER_FIRE_KEY"]]
+        self.change_player_turn_right_key_button.text = "Change Turn Right Key: " + SettingsView.id_to_key[CONFIG["PLAYER_TURN_RIGHT_KEY"]]
+        self.change_player_turn_left_key_button.text = "Change Turn Left Key: " + SettingsView.id_to_key[CONFIG["PLAYER_TURN_LEFT_KEY"]]
+
+    def on_click_change_player_thrust_key(self, event):
```

Comment:
Can't we look at the event to see which button was pressed? Do we really need all these functions?

Reply:
I believe the UIEvent looks something like this:
UIOnClickEvent(source=<arcade.gui.widgets.UIFlatButton object at 0x0000019497ECFBB0>, x=482, y=394)
So no i dont think that is possible